contracts/methodologies/transportation/rideshare/RideshareProject.sol

```diff
@@ -1,36 +1,52 @@
-import "./RideshareToken.sol";
-import "./EmissionsCalc.sol";
-import "openzeppelin-solidity/contracts/math/SafeMath.sol";
+pragma solidity ^0.4.24;
+
 import "openzeppelin-solidity/contracts/ownership/Ownable.sol";
+import "../../abstract/EmissionsStorage.sol";
+import "./RideshareToken.sol";
+
 
 contract RideshareProject is Ownable {
     using SafeMath for uint256;
 
-    RideshareToken public co2kn;
-    EmissionsCalc public ec;
-    uint256 baseline; // Baseline emissions computed off-chain
+    RideshareToken co2kn;
+
+    EmissionsStorage baseline;
+    EmissionsStorage project;
 
+    // Possible to remove all constructor dependencies
+    // and initialize all with new...
     // TODO: Make this an initializable fn for ZOS
     // TODO: Should take an owner as input, not implicitly msg.sender as Ownable forces
     constructor (
-        //address _owner,
-        //RideshareToken _co2kn,
-        EmissionsCalc _ec,
-        uint256 _baseline,
-        uint256 _tokenCap
-    ) {
-        //owner = _owner;
-        //co2kn = _co2kn;
+        RideshareToken _co2kn, // change this to an interface
+        EmissionsStorage _baseline,
+        EmissionsStorage _project
+    ) public {
+
+        co2kn = _co2kn;
         baseline = _baseline;
-        co2kn = new RideshareToken(_tokenCap);
```

Comment:
The reason I had the project create the token is that it needs ownership in order to mint, and ownership is no longer transferable for the mintabletoken in zeppelin (https://github.com/OpenZeppelin/openzeppelin-solidity/blob/master/contracts/token/ERC20/ERC20Mintable.sol).